enhancements/20240820-mtu-and-network-information-propagation.md

```diff
@@ -0,0 +1,318 @@
+```
+
+The above fields are converted to a JSON string and set as a string to k8s CRD NAD object.
+
+```
+cniv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+
+type NetworkAttachmentDefinition struct {
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  Spec NetworkAttachmentDefinitionSpec `json:"spec"`
+}
+
+type NetworkAttachmentDefinitionSpec struct {
+  Config string `json:"config"`
+}
+
+```
+
+Example: The NAD for Harvester storage network.
+
+```
+kind: NetworkAttachmentDefinition
+spec:
+  config: '{"cniVersion":"0.3.1","type":"bridge","dns":{},"bridge":"storage-br","promiscMode":true,"vlan":1,"ipam":{"type":"whereabouts","range":"10.172.0.0/16"}}'
+  
+```
+
+### Harvester UI sets Fields
+
+When configuring [Vlan Network](https://docs.harvesterhci.io/v1.3/networking/harvester-network#vlan-network) from Harvester UI, below fields are touched:
+
+```
+
+Type: `'L2VlanNetwork', 'UntaggedNetwork'`
+Vlan ID:
+ClusterNetwork: User select.  Implicitly converted to `Bridge` name.
+```
+
+:::note
+
+Harvester only exposes very few fields of Bridge CNI to UI.
+
+:::
+
+
+The `Route` related information is converted to an annotation on NAD.
+
+```
+Route:
+  Automatic
+  Manual:
+    CIDR, Gateway
+```
+
+### User Stories
+
+
+#### Story 1: Install a Cluster with the Predefined MTU
+
+The current implementation is OK.
+
+e.g. MTU 9000
+
+After `Clusternetwork` and `Vlanconfig` are created, all following `NADs` will inherit the MTU value 9000. Pod/VM attached to this NAD also has MTU 9000.
+
+#### Story 2: Change the MTU of a Clusternetwork and Vlanconfig on the Fly
+
+The current implementation is not OK.
+
+Step 1: Install the cluster with the default MTU 1500.
+
+After `Clusternetwork` and `Vlanconfig` is created, all following `NAD`s will inherit the MTU value `1500`.
+
+Step 2: Change the MTU to 9000, step by step on each `Vlanconfig`.
+
+Issue/limitation: MTU on the existing `NAD`s are not updated automatically.
+
+#### Story 3: Set the NAD with a different MTU than the Vlanconfig
+
+Step 1: Create a `Vlanconfig` with the default MTU 1500 and one `NAD` on it.
+
+Step 2: Configure the MTU of this `NAD` to a new value 1400 manually.
+
+There is no direct UI.
+
+Issue/limitation: `NAD` does not expose all the fields on UI (including MTU), only via `kubectl`.
+
+#### Story 4: Set the NAD MTU on the Fly
+
+Step 1. Create NAD with MTU 1500.
+
+Step 2. Create POD/VM on the NAD.
+
+Step 3. Change the NAD MTU to 1400.
+
+Step 4. Replace the POD/VM manually.
+
+Issue/limitation: When NAD is changed, the `consumer POD/VM` needs to be replaced manually.
+
+#### Story 5: Set a different MTU on the Guest OS
+
+When a VM starts with the MTU 1500 configured to its nic. The MTU is also 1500 on the related `NAD` and `Vlanconfig`.
+
+What will happen if the MTU is changed to a value like 9000 from the guest OS by the user manually?
```

Comment:
Is it supported officially ? Though user could change it this way, the recommended method is through UI.correct ?

Reply:
In the scenario of IaaS, a user rents a VM from a SP, he may try to modify the MTU from the guest OS for any purpose.

The potential result / affection is unknown yet. 

After investigation, I will update it to Harvester document.